airflow/providers/amazon/aws/waiters/base_waiter.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I thought BatchClient already implements some kind of waiters (I personally never use it in this hook)
- `airflow/providers/amazon/aws/hooks/batch_waiters.json`
- `airflow/providers/amazon/aws/hooks/batch_waiters.py`

Reply:
Yes?  Looks like it.  I've never used that one and it's not something you could inherit and use for other services.

Reply:
I mean a good point one day also migrate batch waiters to generic solution

Reply:
Yeah, this would lay the framework for that and adding whatever other custom waiters that folks want to make.  The more I look at it, the more I think it'll either simplify or replace a lot of Sensors as well, which wasn't the intended purpose but a very nice side effect.